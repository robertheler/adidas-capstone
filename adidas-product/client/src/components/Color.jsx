import React, { Component } from 'react';
import styled from 'styled-components';
import fetchImageLocation from '../helper.js';

const RenderDiv = styled.div`
  margin-top: 20px;
  margin-bottom: 20px;
`;

const StyledImage = styled.div`
  width: 60px;
  height: 60px;
  background: url(${props => props.url});
  background-size: 100%;
  margin: 0 5px;
  opacity: ${props => props.isSelected === 'true' ? "1" : "0.5"};
  box-sizing: border-box;
  border-bottom: ${props => props.isSelected === 'true' ? "3px solid black" : "none"};
  box-sizing: border-box;
  &:hover {
    opacity: 1;
  }
`;

//TODO
const StyledSelectedTick = styled.h1`
`;

class Color extends Component{
  constructor(props){
    super(props);
    this.handleColorChange = this.handleColorChange.bind(this)
  }

  handleColorChange(){
    this.props.handleColorChange(this.props.color)
  }

  render(){
    let url = fetchImageLocation(this.props.product.colors[this.props.color].images[0]);
    return (
      <RenderDiv>
        <StyledImage url={url} onClick={this.handleColorChange} isSelected={this.props.isSelected.toString()}/>
      </RenderDiv>
    );
  }
}

export default Color;