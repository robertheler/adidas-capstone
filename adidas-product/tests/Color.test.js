import React from 'react';
import { shallow, mount, render } from 'enzyme';
import Color from '../client/src/components/Color';
import StyledImage from '../client/src/components/Color';
import sinon from 'sinon';

// json data will be used to generate props
var data = require('../db/data.json');

function setup() {
  const props = {
    color: 0,
    product: data[0],
    isSelected: false,
    handleColorChange: jest.fn()
  };
  const wrapper = shallow(<Color color={props.color} product={props.product} isSelected={props.isSelected} handleColorChange={props.handleColorChange}/>);
  return { wrapper, props };
}

describe('Color Component Test Suite', () => {
  const { wrapper, props } = setup();

  it('It should render', () => {
    expect(wrapper).toHaveLength(1);
  });

  it('It should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('It should trigger the props color change handler', () => {
    wrapper.instance().handleColorChange();
    expect(props.handleColorChange).toHaveBeenCalled();
  });
});