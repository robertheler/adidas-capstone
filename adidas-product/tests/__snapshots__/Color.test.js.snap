// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color Component Test Suite It should render correctly 1`] = `
<styled.div>
  <styled.div
    isSelected="false"
    onClick={[Function]}
    url="https://adidasproducts.s3-us-west-1.amazonaws.com/images/B42200_color0_pic0.jpg"
  />
</styled.div>
`;
